### PR TITLE
XML Transformation failures and suppression

### PR DESCRIPTION
**Pull request by Damovisa**
XML Transform step will now fail on warnings unless the `Octopus.Action.Package.IgnoreConfigTransformationErrors` variable is set to True. Part of #1225.

### CODE REVIEW

File: source/Calamari/Commands/DeployPackageCommand.cs [hunk 1]
```diff
@@ -53,8 +53,8 @@ public override int Execute(string[] commandLineArguments)
             var fileSystem = new CalamariPhysicalFileSystem();
             var scriptEngine = new ScriptEngineSelector();
             var replacer = new ConfigurationVariablesReplacer();
+            var configurationTransformer = new ConfigurationTransformer(variables.GetFlag(SpecialVariables.Package.IgnoreConfigTransformationErrors));
             var substituter = new FileSubstituter();
-            var configurationTransformer = new ConfigurationTransformer();
```
Comment (PaulStovell) [inline source/Calamari/Commands/DeployPackageCommand.cs]: Why change the order these were registered in?

File: source/Calamari/Integration/ConfigurationTransforms/VerboseTransformLogger.cs [hunk 2]
```diff
@@ -86,4 +120,16 @@ public void EndSection(MessageType type, string message, params object[] message
             EndSection(message, messageArgs);
         }
     }
+
+    public delegate void LogDelegate(object sender, ErrorDelegateArgs args);
+
+    public class ErrorDelegateArgs
```
Comment (PaulStovell) [inline source/Calamari/Integration/ConfigurationTransforms/VerboseTransformLogger.cs]: Refactor -> Move to new file :smile:

**Comment by PaulStovell** [comment]
LGTM :+1:, just some minor comments

File: source/Calamari/Commands/DeployPackageCommand.cs (hunk 1 above)
Comment (Damovisa) [inline source/Calamari/Commands/DeployPackageCommand.cs]: Oh, wasn't intentional - I was originally working off an older version that didn't have `variables` defined. I moved it below that then the merge changed it all around. I'll fix :)

File: source/Calamari/Integration/ConfigurationTransforms/VerboseTransformLogger.cs (hunk 2 above)
Comment (Damovisa) [inline source/Calamari/Integration/ConfigurationTransforms/VerboseTransformLogger.cs]: :+1:

**Pull request merged**